libs/ledger/src/chain/block.cpp

```diff
@@ -34,7 +34,7 @@ Block::Block() = default;
 bool Block::operator==(Block const &rhs) const
 {
   // Invalid to compare blocks with no block hash
-  return (!this->body.hash.empty()) && (this->body.hash == rhs.body.hash);
+  return (!this->hash.empty()) && (this->hash == rhs.hash);
```

Comment:
A personal appeal: can this statement be changed into something more compact, like `return !hash.empty() && hash == rhs.hash;`? >_<